src/lib/date/date-formatter.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * @typedef {import('./date.types.js').DateFormat} DateFormat
+ * @typedef {import('./date.types.js').Timezone} Timezone
+ * @typedef {import('./date.types.js').DateFormattedParts} DateFormattedParts
+ * @typedef {import('./date.types.js').DateFormattedPart} DateFormattedPart
+ * @typedef {import('./date.types.js').DatePart} DatePart
+ * @typedef {import('./date.types.js').DateParts} DateParts
+ */
+
+/**
+ * @type {Map<Timezone, Intl.DateTimeFormat>}
+ */
+const DATE_TIME_FORMATS_BY_TIMEZONE = new Map();
+function getDateTimeFormat (timezone) {
+  let format = DATE_TIME_FORMATS_BY_TIMEZONE.get(timezone);
+  if (format == null) {
+    format = new Intl.DateTimeFormat('en', {
+      year: 'numeric',
+      month: '2-digit',
+      day: '2-digit',
+      hour: '2-digit',
+      minute: '2-digit',
+      second: '2-digit',
+      fractionalSecondDigits: 3,
+      hour12: false,
+      timeZoneName: 'longOffset',
+      timeZone: (timezone === 'local') ? undefined : timezone,
+    });
+    DATE_TIME_FORMATS_BY_TIMEZONE.set(timezone, format);
+  }
+  return format;
+}
```

Comment:
Because this is a heavy computation (date API sucks). And the `Intl.DateTimeFormat` by timezone can be reused for every instances of `DateFormatter`.

I'll add a comment about that.